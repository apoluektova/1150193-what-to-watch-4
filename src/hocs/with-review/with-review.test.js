import React from "react";
import renderer from "react-test-renderer";
import withReview from "./with-review.js";

const MOVIE = {
  id: 1,
  previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  previewVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  title: `The Grand Budapest Hotel`,
  backgroundImage: `img/bg-the-grand-budapest-hotel.jpg`,
  backgroundColor: `#A6B7AC`,
  poster: `https://loremflickr.com/cache/resized/65535_49824720108_18907b74af_z_273_410_nofilter.jpg`,
  genre: `Drama`,
  releaseDate: 2014,
  description: `The Grand Budapest Hotel is a 2014 comedy-drama film written and directed by Wes Anderson, which explores tragedy, war, fascism, nostalgia, friendship, and loyalty.`,
  rating: {
    score: 9.3,
    count: 250,
  },
  director: `Wes Anderson`,
  actors: [`Michael Fassbender`, `Marion Cotillard`, `Paddy Considine`, `Sean Harris`],
  runtime: 99,
  isFavorite: false,
};

const MockComponent = (props) => {
  return (
    <div {...props} />
  );
};

const MockComponentWrapped = withReview(MockComponent);

it(`withReview should render correctly`, () => {
  const tree = renderer
     .create(<MockComponentWrapped
       movie={MOVIE}
       onRatingChange={() => {}}
       onReviewChange={() => {}}
       onReviewFormSubmit={() => {}}
       isSubmitButtonDisabled={false}
       onReviewSubmit={() => {}}
     />)
     .toJSON();

  expect(tree).toMatchSnapshot();
});

