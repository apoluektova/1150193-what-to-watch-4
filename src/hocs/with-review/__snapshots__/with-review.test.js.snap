// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withReview should render correctly 1`] = `
<div
  isSubmitButtonDisabled={true}
  movie={
    Object {
      "actors": Array [
        "Michael Fassbender",
        "Marion Cotillard",
        "Paddy Considine",
        "Sean Harris",
      ],
      "backgroundColor": "#A6B7AC",
      "backgroundImage": "img/bg-the-grand-budapest-hotel.jpg",
      "description": "The Grand Budapest Hotel is a 2014 comedy-drama film written and directed by Wes Anderson, which explores tragedy, war, fascism, nostalgia, friendship, and loyalty.",
      "director": "Wes Anderson",
      "genre": "Drama",
      "id": 1,
      "isFavorite": false,
      "poster": "https://loremflickr.com/cache/resized/65535_49824720108_18907b74af_z_273_410_nofilter.jpg",
      "previewImage": "img/fantastic-beasts-the-crimes-of-grindelwald.jpg",
      "previewVideo": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
      "rating": Object {
        "count": 250,
        "score": 9.3,
      },
      "releaseDate": 2014,
      "runtime": 99,
      "title": "The Grand Budapest Hotel",
      "videoLink": "https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4",
    }
  }
  onRatingChange={[Function]}
  onReviewChange={[Function]}
  onReviewFormSubmit={[Function]}
  onReviewSubmit={[Function]}
/>
`;
